File: java_old/game-online/matchmaker/src/main/java/ru/atom/matchmaker/players/model/Player.java
package ru.atom.matchmaker.players.model;

import javax.persistence.*;
import java.io.Serializable;
//import java.util.List;

@Entity
@Table(name = "players", schema = "public")
public class Player implements Serializable {
    @Id
    @Column(name = "id", updatable = false, nullable = false)
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private long id;

    @Column(name = "login", unique = true, nullable = false, length = 40)
    private String login;

    @Column(name = "password", nullable = false, length = 40)
    private String password;

    @Column(name = "rating", nullable = false)
    private int rating;

    @Column(name = "isonline")
    private boolean isOnline;

    @Column(name = "insearch")
    private boolean inSearch;

    @Column(name = "idsession")
    private long idSession;

    public long getId() {

        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public boolean isOnline() {
        return isOnline;
    }

    public void setOnline(boolean online) {
        isOnline = online;
    }

    public boolean isInSearch() {
        return inSearch;
    }

    public void setInSearch(boolean inSearch) {
        this.inSearch = inSearch;
    }

    public long getIdSession() {
        return idSession;
    }

    public void setIdSession(long idSession) {
        this.idSession = idSession;
    }

    public int getRating() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    public Player() {
        rating = 1500;
    }

    public Player(String login, String password) {
        this.login = login;
        this.password = password;
        this.rating = 1500;
    }

    @Override
    public String toString() {
        return "Player(" + "id =" + id + ", name ='" + login
                + '\'' + ", rating = "
                + rating + ')';
    }
}

File: kotlingirl-registry/src/main/kotlin/com/kotlingirl/registry/model/Player.kt
package com.kotlingirl.registry.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "player", schema = "game")
@JsonIgnoreProperties("password")
data class Player(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long,

        @Column(name = "games_played")
        var gamesPlayed: Int = 0,

        @Column(name = "games_won")
        var gamesWon: Int = 0,

        @Column(name = "login")
        var login: String,

        @Column(name = "password")
        var password: CharArray,

        @Column(name = "rating")
        var rating: Int = 1000,

        @Column(name = "modified_at")
        var modifiedAt: Date = Date(),

        @Column(name = "registration_date")
        var registrationDate: Date? = Date(),

        @Column(name = "logged")
        var logged: Boolean = false,

/*        @ManyToOne
        @JoinColumn(name = "sessiondata_id", insertable = false, updatable = false)*/
        @Column(name = "current_game")
        var currentGame: Long? = null
)
{
    constructor(login: String, password: CharArray) : this(0,0,0,login,password,1000, Date(), Date(), false, null)

        override fun equals(other: Any?): Boolean {
                if (this === other) return true
                if (javaClass != other?.javaClass) return false

                other as Player

                if (id != other.id) return false
                if (gamesPlayed != other.gamesPlayed) return false
                if (gamesWon != other.gamesWon) return false
                if (login != other.login) return false
                if (!password.contentEquals(other.password)) return false
                if (rating != other.rating) return false
                if (modifiedAt != other.modifiedAt) return false
                if (registrationDate != other.registrationDate) return false
                if (logged != other.logged) return false
                if (currentGame != other.currentGame) return false

                return true
        }

        override fun hashCode(): Int {
                var result = id.hashCode()
                result = 31 * result + gamesPlayed
                result = 31 * result + gamesWon
                result = 31 * result + login.hashCode()
                result = 31 * result + password.contentHashCode()
                result = 31 * result + rating
                result = 31 * result + modifiedAt.hashCode()
                result = 31 * result + (registrationDate?.hashCode() ?: 0)
                result = 31 * result + logged.hashCode()
                result = 31 * result + (currentGame?.hashCode() ?: 0)
                return result
        }
}